hotsos/core/ycheck/engine/properties/requires/types/apt.py

```diff
@@ -28,32 +28,147 @@ def installed_versions(self):
 
         return _versions
 
-    def package_version_within_ranges(self, pkg, versions):
-        result = False
-        versions = sorted(versions, key=lambda i: str(i['min']), reverse=True)
+    def normalize_version_criteria(self, version_criteria):
+        """Normalize all the criterions in a criteria.
+
+        Normalization does the following:
+        - removes empty criteria
+        - replaces old ops with the new ones
+        - sorts each criterion(ascending) and criteria(descending)
+        - adds upper/lower bounds to criteria, where needed
+
+        Args:
+            version_criteria (list): List of version ranges to normalize
+
+        Returns:
+            list: Normalized list of version ranges
+        """
+
+        # Step 0: Remove empty criteria
+        version_criteria = [x for x in version_criteria if len(x) > 0]
+
+        # Step 1: Replace legacy ops with the new ones
+        legacy_ops = {"min": "ge", "max": "le"}
+        for idx, version_criterion in enumerate(version_criteria):
+            for lop, nop in legacy_ops.items():
+                if lop in version_criterion:
+                    version_criterion[nop] = version_criterion[lop]
+                    del version_criterion[lop]
+                    version_criteria[idx] = version_criterion
+
+        # Step 2: Sort each criterion in itself, so the smallest version
+        # appears first
+        for idx, version_criterion in enumerate(version_criteria):
+            version_criterion = dict(sorted(version_criterion.items(),
+                                     key=lambda item: str(item[1])))
```

Comment:
I know this existing code is string sorting and that is my bad but it is in fact not always safe for package versions e.g.

```
$ dpkg --compare-versions 1.2 gt 1.2~1 && echo "yes"
yes
$ echo "print('1.2' > '1.2~1')"| python3 
False
```

Reply:
maybe it is ok for now but need think about when this will get us into trouble and whether it is worth fixing now